onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -308,24 +309,28 @@ def destroy(self, request, *args, **kwargs):
 
                 if not instance_ids:
                     raise ParseError(_(u"Invalid data ids were provided."))
-
                 initial_count = self.object.submission_count()
-                queryset = Instance.objects.filter(
+
+                Instance.objects.filter(
                     id__in=instance_ids,
                     xform=self.object,
                     # do not update this timestamp when the record have
                     # already been deleted.
                     deleted_at__isnull=True
-                )
-                # loop through queryset
-                # then call delete_instance that calls .save()
-                # to allow emitting post_save signal
-                for instance in queryset_iterator(queryset):
-                    delete_instance(instance, request.user)
+                ).update(deleted_at=timezone.now(), deleted_by=request.user)
+
                 # updates the num_of_submissions for the form.
                 after_count = self.object.submission_count(force_update=True)
                 number_of_records_deleted = initial_count - after_count
 
+                # update the date modified field of the project
+                self.object.project.date_modified = timezone.now()
```

Comment:
You need to call the save function after this

Reply:
done